### PR TITLE
Lsm6ds refactor

### PR DESCRIPTION
**Pull request by siddacious**
This refactor makes subclasses for each sensor and pulls unique code into each subclass

### CODE REVIEW

File: adafruit_lsm6ds.py [hunk 1]
```diff
@@ -163,74 +165,73 @@ class LSM6DSOX: #pylint: disable=too-many-instance-attributes
...
-        self._accel_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
-        self._gyro_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+        self.accelerometer_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+        self.gyro_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+
+        self.accelerometer_range = AccelRange.RANGE_4G #pylint: disable=no-member
+        self.gyro_range = GyroRange.RANGE_250_DPS #pylint: disable=no-member
 
-        self._accel_range = AccelRange.RANGE_4G #pylint: disable=no-member
         self._cached_accel_range = self._accel_range
```
Comment (ladyada) [inline adafruit_lsm6ds.py]: why is this line here?

File: adafruit_lsm6ds.py [hunk 2]
```diff
@@ -163,74 +165,73 @@ class LSM6DSOX: #pylint: disable=too-many-instance-attributes
...
-        self._gyro_range = GyroRange.RANGE_250_DPS #pylint: disable=no-member
         self._cached_gyro_range = self._gyro_range
```
Comment (ladyada) [inline adafruit_lsm6ds.py]: ditto

File: examples/lsm6ds33_simpletest.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import busio
+from adafruit_lsm6ds import LSM6DS33
+
+i2c = busio.I2C(board.SCL, board.SDA)
+
+sensor = LSM6DS33(i2c)
+
+while True:
+    print("Acceleration: X:%.2f, Y: %.2f, Z: %.2f m/s^2"%(sensor.acceleration))
+    print("Gyro X:%.2f, Y: %.2f, Z: %.2f degrees/s"%(sensor.gyro))
```
Comment (ladyada) [inline examples/lsm6ds33_simpletest.py]: add spaces between %:

`print("Gyro X:%.2f, Y: %.2f, Z: %.2f degrees/s" % (sensor.gyro))`

File: examples/lsm6ds_full_test.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+sensor = LSM6DS33(i2c)
+#sensor = LSM6DSOX(i2c)
+#sensor = ISM330DHCT(i2c)
+
+sensor.accelerometer_range = AccelRange.RANGE_8G
+print("Accelerometer range set to: %d G"%AccelRange.string[sensor.accelerometer_range])
+
+sensor.gyro_range = GyroRange.RANGE_2000_DPS
+print("Gyro range set to: %d DPS"%GyroRange.string[sensor.gyro_range])
```
Comment (ladyada) [inline examples/lsm6ds_full_test.py]: ditto for all % strings, add a space between % and left/right halves for readability

File: adafruit_lsm6ds.py (hunk 1 above)
Comment (siddacious) [reply adafruit_lsm6ds.py]: I usually have lines like these to set the default values to something more useful, though 250dps  should change then as it's already the default.

Comment (ladyada) [reply adafruit_lsm6ds.py]: ` self._cached_accel_range = self._accel_range`
should be handled within the property/setter right?

File: examples/lsm6ds_full_test.py (hunk 4 above)
Comment (siddacious) [reply examples/lsm6ds_full_test.py]: done

File: adafruit_lsm6ds.py (hunk 1 above)
Comment (siddacious) [reply adafruit_lsm6ds.py]: yes, you're right. I misunderstood

Comment (siddacious) [reply adafruit_lsm6ds.py]: Pylint just reminded me that it doesn't like when you do that because it ends up getting defined outside of `__init__`
image

Comment (ladyada) [reply adafruit_lsm6ds.py]: set it to None at the top of __init__

**Comment by ladyada** [comment]
can merge once passing
